### PR TITLE
overlord/snapstate: track snaps in state

### PR DESCRIPTION
**Pull request by niemeyer**

### CODE REVIEW

File: overlord/snapstate/snapmgr.go [hunk 1]
```diff
@@ -60,6 +61,14 @@ type SnapSetup struct {
 	SnapPath string `json:"snap-path"`
 }
 
+// snapState holds the state for a snap installed in the system.
+type snapState struct {
+	Sequence []*snap.SideInfo `json:"sequence"` // Last is current
+	Active   bool             `json:"active,omitempty"`
+	Channel  string           `json:"channel,omitempty"`
+	DevMode  bool             `json:"dev-mode,omitempty"`
```
Comment (zyga) [inline overlord/snapstate/snapmgr.go]: Goodness! :)

Comment (zyga) [inline overlord/snapstate/snapmgr.go]: Does it make sense to track connections here?

**Comment by pedronis** [comment]
LGTM

File: overlord/snapstate/snapmgr.go (hunk 1 above)
Comment (niemeyer) [inline overlord/snapstate/snapmgr.go]: Will respond on the channel.

**Pull request merged**